lib/ddtrace/contrib/sidekiq/tracing.rb

```diff
@@ -18,6 +18,8 @@ def initialize(options = {})
         def job_resource(job)
           if job['wrapped']
             job['wrapped']
+          elsif job['class'] == 'Sidekiq::Extensions::DelayedClass'
+            YAML.load(job['args'].first)[0..1].join('.') rescue job['class'] # rubocop:disable Security/YAMLLoad
```

Comment:
Looks like `YAML.load` is dangerous. Can you use `safe_load` instead?

Reply:
It's possible for `YAML.load` to fail if queued jobs are out of sync with loaded code, or for it to be unsafe if jobs were queued by untrusted sources, but I don't think it's possible to swap in `safe_load` here because we can't constrain the classes that these jobs may refer to. Queueing work for arbitrary classes is the power of Sidekiq's delayed extensions, and Sidekiq's [own code](https://github.com/mperham/sidekiq/blob/148afbf0779b246d82a95fa1b6b984c67ebf6229/lib/sidekiq/api.rb#L429) uses `load` in that same unsafe way (if I understand it correctly).

I totally understand if you think this convenience isn't worth the potential risk or the rubocop exception. I wasn't sure of a better solution though.

Reply:
The built-in YAML parser, `psych`, doesn't like parsing that YAML payload in safe mode, like @joeyAghion mentioned:
```ruby
[1] pry(main)> require 'yaml'
[2] pry(main)> YAML.safe_load(File.read('sidekiq_delayed_class.yaml'))
Psych::DisallowedClass: Tried to load unspecified class: ClientTracerTest::DelayableClass
from ~/.rbenv/versions/2.5.1/lib/ruby/2.5.0/psych/class_loader.rb:97:in `find'
```

But `safe_yaml` works correctly:
```ruby
[1] pry(main)> require 'safe_yaml'
[2] pry(main)> YAML.safe_load(File.read('sidekiq_delayed_class.yaml'))
=> ["ClientTracerTest::DelayableClass", ":do_work", []]
```

For context, `safe_yaml` is a popular gem (with as many downloads as `rubocop` from rubygems, for reference), without any transient dependencies. But including a new gem as dependency to `dd-trace-rb` is always a concern.

Reply:
Hmmm, yeah, having to add a dependency like this is not great, especially if `gem 'safe_yaml'` might interfere with users' applications which use `YAML`: we wouldn't want tracing to suddenly swap out their YAML parsing behavior.

I'm open to other suggestions, but I don't think changing out `YAML` or using a function that exposes arbitrary code execution is a good idea. Is there another way to do this?

Reply:
Yeah this is a tough one. In our code base jobs are "trusted" but I could imagine other consumers not wanting to adopt that assumption.

I'll close this PR. Thanks for the feedback in any case!

If there's any interest, I could fashion our patch into a tiny extension gem that other users of delayed extensions could choose to include.

Reply:
Alright, bummer that there isn't any other way (was hoping maybe we could regex it or something), but feel free to make a small extension gem!

Reply:
I considered regex or other string-slicing but expect it would be uglier and depend on even more assumptions. Something like:

```ruby
job['args'].first.match(/^---\n- !ruby\/class '(?<class>.*)'\n- :(?<method>.*)\n-/m) # => #<MatchData "..." class:"String" method:"to_s">
```

I'd rather not add this imperfect YAML-parsing to the library, personally.